src/kernels/raw/finder/preferredRemoteKernelIdProvider.ts

```diff
@@ -38,7 +38,15 @@ export class PreferredRemoteKernelIdProvider {
         }
     }
 
-    public async storePreferredRemoteKernelId(uri: Uri, id: string | undefined): Promise<void> {
+    public async clearPreferredRemoteKernelId(uri: Uri): Promise<void> {
+        await this.storePreferredRemoteKernelIdInternal(uri);
```

Comment:
Bug fix, added tests:
* Assume you open a notebook and select a remote kernel
* Now run a cell
* Open the same notebook tomorrow, it will connect to the same remote kernel (good)
* Shutdown the remote server
* Open the notebook, you'll get an error about the server not being available, good
* Now switch the kernel to a local kernel
* Close vsc & open the same notebook again & it will try to connect to the same remote server.

We weren't clearing the mapping.